functions/localhost_or_hosts_with_pe_profile.pp

```diff
@@ -12,21 +12,26 @@
 function puppet_metrics_dashboard::localhost_or_hosts_with_pe_profile(
   String $profile,
 ) >> Array {
-  if $settings::storeconfigs {
-    $_profile = capitalize($profile)
-    $hosts = puppetdb_query("resources[certname] {
-      type = 'Class' and
-      title = 'Puppet_enterprise::Profile::${_profile}' and
-      nodes { deactivated is null and expired is null }
-    }").map |$nodes| { $nodes['certname'] }
+  if (versioncmp($facts['pe_server_version'], '2019.5') >= 1 ) and ($profile == 'puppetdb') {
```

Comment:
For this Travis error:

```
Error: Evaluation Error: Error while evaluating a Function Call, 'versioncmp' parameter 'a' expects a String value, got Undef (file: /etc/puppetlabs/code/modules/puppet_metrics_dashboard/functions/localhost_or_hosts_with_pe_profile.pp, line: 15, column: 7) on node centos-7-x64.c.travis-ci-prod-2.internal
```
Maybe:

```
if ($facts['pe_server_version'] =~ NotUndef and versioncmp($facts['pe_server_version'], '2019.5') >= 1 ) and ($profile == 'puppetdb') {
```

See also:

https://github.com/puppetlabs/puppetlabs-puppet_metrics_collector/blob/master/functions/version_based_excludes.pp

https://github.com/puppetlabs/puppetlabs-puppet_metrics_collector/blob/master/functions/hosts_with_pe_profile.pp

Reply:
Testing w a GitHub commit now